src/pages/studyView/charts/barChart/CustomBinsModal.tsx

```diff
@@ -52,6 +76,22 @@ export default class CustomBinsModal extends React.Component<
         return customBinsAreValid(this.newStringBins);
     }
 
+    changeBinsCheckbox(option: BinMethodOption) {
+        this.props.store.binMethod = option;
+    }
+
+    @action
+    updateBinSize(value: number) {
+        this.currentGenerateBins.binSize = value;
+        this.props.store.generateBinsConfig.binSize = value;
+    }
+
+    @action
+    updateAnchorValue(value: number) {
+        this.currentGenerateBins.anchorValue = value;
+        this.props.store.generateBinsConfig.anchorValue = value;
+    }
```

Comment:
Why is this @action not handled by the updateCustomBins callback?

Reply:
Updated the code, but not as part of updateCustomBins, in order to see bin size and anchor value in the input fields as they are being typed. Moved only the store props to updateCustomBins.